tree_labeller/tree/selectors/sampling.py

```diff
@@ -55,6 +55,7 @@ def calc_prob(tree: NodeMixin):
 
 
 def weighted_sample_without_replacement(population, weights, k=1):
+    k = min(len(weights), k)
```

Comment:
To behave more like [`numpy.random.choice`](https://numpy.org/doc/stable/reference/random/generated/numpy.random.choice.html):

```suggestion
    assert len(population) == len(weights)
    assert k >= len(population)
```
and 
```python
    k = min(len(population), k)
```
should be on the calling method side.

Reply:
fixed